### PR TITLE
Implement ApiClient for all endpoints

### PR DESCRIPTION
**Pull request by mads256h**
Change delete endpoints to only take the id of the object to delete. Change delete endpoints to use the HTTP DELETE method.

Add protocol objects to the frontend for talking to the backend. Use kotlin coroutines instead of callbacks for the ApiClient.

Create tests that connect to http://localhost:3000 and tests with the backend.

### CODE REVIEW

File: backend/src/protocol/devices.rs [hunk 1]
```diff
@@ -4,3 +4,8 @@ use serde::{Deserialize, Serialize};
 pub struct CreateDeviceRequest {
     pub effect: f64,
 }
+
+#[derive(Deserialize, Serialize)]
+pub struct DeleteDeviceRequest {
+    pub id: i64,
```
Comment (Mikkel-Boje) [inline backend/src/protocol/devices.rs:10]: Is this the device' id?

File: backend/src/protocol/tasks.rs [hunk 2]
```diff
@@ -8,3 +8,8 @@ pub struct CreateTaskRequest {
     pub duration: Milliseconds,
     pub device_id: i64,
 }
+
+#[derive(Deserialize, Serialize)]
+pub struct DeleteTaskRequest {
+    pub id: i64,
```
Comment (Mikkel-Boje) [inline backend/src/protocol/tasks.rs:14]: Is this the tasks' id?

**Review by Mikkel-Boje — APPROVED** [review]
Axum is amazing 😄

**Review by S3JER — APPROVED** [review]

File: .github/workflows/frontend.yml [hunk 3]
```diff
@@ -30,8 +34,17 @@ jobs:
...
+        run: |
+          cargo install sqlx-cli --no-default-features --features sqlite
+          cargo sqlx db create
+          cargo sqlx migrate run
+          cargo build
+          cargo run &
+        working-directory: backend
       - name: Build with Gradle
-        run: ./gradlew build
+        run: ./gradlew build -x test
```
Comment (Darkros1245) [inline .github/workflows/frontend.yml:47]: Hvorfor kører vi ikke test, men køre det lige bagefter som en seperat task?

File: backend/src/protocol/devices.rs (hunk 1 above)
Comment (mads256h) [reply backend/src/protocol/devices.rs:10]: Yes it matches the name in the device struct

File: backend/src/protocol/tasks.rs (hunk 2 above)
Comment (mads256h) [reply backend/src/protocol/tasks.rs:14]: Yes it matches the name in the task struct

File: frontend/app/src/test/java/dk/scheduling/schedulingfrontend/ApiServiceTest.kt [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            assert(fullTasks.size == 1)
+
+            val gottenTask = fullTasks.single()
+
+            assertEquals(task, gottenTask)
+        }
+    }
+
+    private fun createAccount(username: String): Pair<UUID, String> {
+        return runBlocking {
```
Comment (Darkros1245) [inline frontend/app/src/test/java/dk/scheduling/schedulingfrontend/ApiServiceTest.kt:96]: Skal den ikke markeres som et label, for at man kan jump til den?

**Review by Darkros1245 — APPROVED** [review]

**Pull request merged**